libqtile/utils.py

```diff
@@ -199,6 +200,21 @@ def get_cache_dir() -> str:
         os.makedirs(cache_directory)
     return cache_directory
 
+def _get_config_file():
+    config_home = os.getenv("XDG_CONFIG_HOME", "~/.config")
+    if (config_file:= Path(f"{config_home}/qtile/config.py")).exists():
+        return config_file
+
+    xdg_config_dirs = os.getenv("XDG_CONFIG_DIRS", "/etc/xdg/").split(":")
+    for config_dir in xdg_config_dirs:
+        if (system_wide_config := Path(f"{config_dir}/qtile/config.py")).exists():
```

Comment:
These also might need the tildes resolved in each loop. I'm basing this only on i3's code though so if you've tested already and it works then it's fine to leave it

Reply:
This would in fact break for any directory in `$XDG_CONFIG_DIRS` with a `~`

Fixed here: https://github.com/qtile/qtile/pull/4490/commits/8b38ef38099d77b3bb96fb7d76e9832c7d724b4e